### PR TITLE
Improve buy sell

### LINKED ISSUE #65: If user is not connected, we should not let him proceed with enabling asset
Currently:
Screenshot 2020-03-24 at 22 50 22

Should look like:
Screenshot 2020-03-24 at 22 51 19

Design specification:
https://www.figma.com/file/Cirn0tIPXvaDb42dX9XUAq/Playground?node-id=671%3A44

### LINKED ISSUE #66: Signing transaction circle should actually be a loading indicator
Currently:
Screenshot 2020-03-24 at 22 52 55

Should:
Screenshot 2020-03-24 at 22 53 44

Figma design specification:
https://www.figma.com/file/Cirn0tIPXvaDb42dX9XUAq/Playground?node-id=153%3A771

### LINKED ISSUE #67: Unconfirmed Transaction indicator only showed up once the transaction got mined
Just did my first test buy and it seems like the UI is not getting updated once I´ve signed the transaction:

Screenshot 2020-03-24 at 22 55 31

The UI should immediately switch to the unconfirmed state once the user signed the transaction!

### PR DESCRIPTION
**Pull request by AugustoL**
fix #65 - ONly the button was added, we have to connect it to the connect function, which is in the components. @jpkcambridge maybe makes sense to move all the web3 connection logic to the store and execute the connect function form there? To dont have it repeated in the code.
fix #66 #68 - Loader component with a message received by props added. It can be used in the rest of the app.
fix #67 - I think I fixed the states of the txs, should be fixed now.

### CODE REVIEW

File: src/components/Buy/BuySign.js [hunk 1]
```diff
@@ -75,21 +70,8 @@ class BuySign extends React.Component {
...
       <FormWrapper>
-        <InfoRow>
-          <FormInfoText>Price</FormInfoText>
-          <div>{price} {collateralType}</div>
-        </InfoRow>
-        <InfoRow>
-          <FormInfoText>{infotext}</FormInfoText>
-          <div>{priceToBuy} {collateralType}</div>
-        </InfoRow>
-        <InfoRow>
```
Comment (jpkcambridge) [inline src/components/Buy/BuySign.js]: Why are we removing the Price/Pay Amount/Receive numbers?

Comment (AugustoL) [reply src/components/Buy/BuySign.js]: In the designs this is not showed when is not connected. And thats the section that appears when is not conencted.

Comment (jpkcambridge) [reply src/components/Buy/BuySign.js]: Oh, I thought this was the section that shows while you are signing a transaction for a buy

**Comment by jpkcambridge** [comment]
@AugustoL Regarding #65 Totally refactoring the connect features.  And good point regarding moving the logic to the store

**Comment by jpkcambridge** [comment]
@AugustoL I know you said it's only the button right now, but when I do connect, it stays just on the view with the "Connect Wallet" button, which means the rest of the BuySell can't be used.  Until the logic exists to get past that screen, we shouldn't merge this.  I think a quick way to get past that is on connect to set the state such that the next screen appears.

**Comment by AugustoL** [comment]
Ups! good catch @jpkcambridge ! Should be fixed now

**Comment by pimato** [comment]
This view should only be there for the enabling flow and not for the buy/sell process:

Screenshot 2020-03-27 at 08 14 17

instead we should keep the order informations, like we have defined in the Figma design:

Screenshot 2020-03-27 at 08 16 26

**Comment by pimato** [comment]
Also could we only get the "connect to proceed view" for the sell side **IF** the user needs to enable the DXD token?

this:
Screenshot 2020-03-27 at 08 18 18

should be replaced with this:
Screenshot 2020-03-27 at 08 19 07

the reason is we want to give even disconnected users the chance to play around with the order form.
Screenshot 2020-03-27 at 08 19 07

**Comment by pimato** [comment]
If a cancel my metamask transaction the UI is still thinking I am gonna sign the transaction instead the UI should detect the cancelation and show this:

Screenshot 2020-03-27 at 08 35 12

For design reference please see the [figma specification](https://www.figma.com/file/Cirn0tIPXvaDb42dX9XUAq/Playground?node-id=727%3A400)!

**Comment by jpkcambridge** [comment]
Bigger refactor going on to bring in typescript and some other infrastructure.  Closing this for now and will port these changes to new PRs after the typescript/infrastructure refactor.

**Pull request closed without merging**